case_studies/CARD/Bug-Drug Code.R

```diff
@@ -0,0 +1,154 @@
+# config.R
+url <- "https://card.mcmaster.ca/download/0/broadstreet-v3.3.0.tar.bz2"
+destfile <- "broadstreet-v3.3.0.tar.bz2"
+
+# Download the file
+download.file(url, destfile)
+
+#Extract the file
+install.packages("R.utils")
+library(R.utils)
+
+# Decompress the file
+bunzip2("broadstreet-v3.3.0.tar.bz2", destname = "broadstreet-v3.3.0.tar", remove = FALSE)
+file.rename("broadstreet-v3.3.0.tar", "broadstreet-v3.3.0_old.tar")
+
+# Extract the tar file
+untar("broadstreet-v3.3.0_old.tar", exdir = "CARD_data")
+
+# List the contents of the extraction directory
+list.files("CARD_data")
+
+# Parse the ARO_index.tsv file using read.delim
+aro_index <- read.delim("CARD_data/ARO_index.tsv", header = TRUE, sep = "\t")
+
+
+# Map CARD Short Name
+# Load necessary library
+library(dplyr)
+
+#  Read the files
+aro_index <- read.delim("CARD_data/aro_index.tsv", sep = "\t", header = TRUE)
+antibiotics_data <- read.delim("CARD_data/shortname_antibiotics.tsv", sep = "\t", header = TRUE)
+pathogens_data <- read.delim("CARD_data/shortname_pathogens.tsv", sep = "\t", header = TRUE)
+
+
+# Mutate data
+aro_index <- aro_index %>%
+  mutate(
+    pathogen = sapply(strsplit(CARD.Short.Name, "_"), `[`, 1),   # First part: Pathogen
+    gene = sapply(strsplit(CARD.Short.Name, "_"), `[`, 2),       # Second part: Gene
+    drug = ifelse(sapply(strsplit(CARD.Short.Name, "_"), length) == 3,   # Third part: Drug
+                  sapply(strsplit(CARD.Short.Name, "_"), `[`, 3), NA),
+    Protein.Accession = Protein.Accession   # Include existing Protein.Accession column
+  )
+
+# View the mutated data
+head(aro_index)
+
+
+# Extract pathogen, gene, drug, and include Protein.Accession from 'CARD.Short.Name'
+aro_index_clean <- aro_index %>%
+  mutate(
+    pathogen = sapply(strsplit(CARD.Short.Name, "_"), `[`, 1),  # Extract pathogen
+    gene = sapply(strsplit(CARD.Short.Name, "_"), `[`, 2),      # Extract gene
+    drug = ifelse(sapply(strsplit(CARD.Short.Name, "_"), length) == 3,   # Extract drug
+                  sapply(strsplit(CARD.Short.Name, "_"), `[`, 3), NA),
+    Protein.Accession = Protein.Accession  # Include the Protein.Accession column
+  )
+
+# Merge aro_index_clean with the antibiotics_data and pathogens_data
+# For merging with antibiotics_data
+merged_data_antibiotics <- left_join(aro_index_clean, antibiotics_data,
+                                     by = c("drug" = "AAC.Abbreviation"))
+
+# For merging with pathogens_data
+merged_data_pathogens <- left_join(merged_data_antibiotics, pathogens_data,
+                                   by = c("pathogen" = "Abbreviation"))
+
+# View the resulting merged data
+head(merged_data_pathogens)
+
+
+#filter out rows where pathogen is empty
+cleaned_data <- merged_data_pathogens %>%
+  distinct() %>%
+  filter(!is.na(Pathogen)) # Use 'Pathogen' instead of 'pathogen'
+View(cleaned_data)
+
+# Group by Pathogen, Gene, Drug, and Protein.Accession, then summarize Antibiotic information
```

Comment:
I notice in summarized data that some AMR mechanisms or drug classes are missing. For your specific bug-drug combo, this doesn't seem to have any impacts, but **if** we want to generalize this script, using `$Drug.Class`, `$Resistance.Mechanism` and/or `$AMR.Gene.Family` to better extract these data. 

For example, in `cleaned_data`, if you look for gene _acrA_, the `$CARD.Short.Name` will show multidrug resistance for the _E. coli_ _acrR_ gene (in complex with AcrA) with Ecol_AcrR_MULT. For _E. coli_ _acrA_ directly above it, the `$CARD.Short.Name` field is Ecol_acrA, no MULT. But looking under `$Drug.Class` or `$Resistance.Mechanism` shows it's the same efflux mechanism that affects many different drug classes.